src/components/scenes/SendConfirmationScene.js

```diff
@@ -316,14 +326,35 @@ export class SendConfirmation extends Component<Props, State> {
     }
   }
 
-  onExchangeAmountChanged = ({ nativeAmount, exchangeAmount }: ExchangedFlipInputAmounts) => {
+  onExchangeAmountChanged = async ({ nativeAmount, exchangeAmount }: ExchangedFlipInputAmounts) => {
+    const { fiatPerCrypto, coreWallet, sceneState, currencyCode, isLimitExceeded, newSpendInfo, updateTransaction } = this.props
     this.setState({
       showSpinner: true,
       nativeAmount,
       exchangeAmount
     })
-
-    this.props.updateAmount(nativeAmount, exchangeAmount, this.props.fiatPerCrypto.toString())
+    const count = ++this.count
+    const amountFiatString: string = bns.mul(exchangeAmount, fiatPerCrypto.toString())
+    const amountFiat: number = parseFloat(amountFiatString)
+    const metadata: EdgeMetadata = { amountFiat }
+    const guiMakeSpendInfo = { nativeAmount, metadata }
+
+    const guiMakeSpendInfoClone = { ...guiMakeSpendInfo }
+    const spendInfo = getSpendInfoWithoutState(guiMakeSpendInfoClone, sceneState, currencyCode)
+
+    try {
+      newSpendInfo(spendInfo, isLimitExceeded)
+      const edgeTransaction = await makeSpend(coreWallet, spendInfo)
+      if (count === this.count) {
+        this.lastSeenCount = count
+        updateTransaction(edgeTransaction, guiMakeSpendInfoClone, false, null)
+        this.setState({ showSpinner: false })
+      } else if (count > this.lastSeenCount) {
```

Comment:
This is the case where the core is lagging the GUI, but still making valid progress. We might consider updating the fees still in this case, but I agree that just ignoring the outdated transaction is a valid option too.